Add `strictTypes` option to `regexp/no-useless-flag` rule (#292)
ota-meshi/eslint-plugin-regexp: docs/rules/no-useless-flag.md

{
  "regexp/no-useless-flag": ["error",
    {
      "ignore": [], // An array of "i", "m", "s", "g" and "y".
      "strictTypes": true
    }
  ]
}
```

- `ignore` ... An array of flags to ignore from the check.
- `strictTypes` ... If `true`, strictly check the type of object to determine if the regex instance was used in `search()` and `split()`. Default is `true`. This option is only effective for verifying the `g` and `y` flags.  

>   You cannot off of this option if you are using TypeScript.

Reviewer: ```suggestion
  This option is always on when using TypeScript.
```

Author: Thank you!
